lib/spec.go

```diff
@@ -240,25 +287,44 @@ func ReadSpec(filename string, schemas *terraform.Schemas) (*Spec, tfdiags.Diagn
 }
 
 // ParseSpec parses the spec contained in the []byte parameter and returns the resulting Spec or a Diagnostics if error occured in the process
-func ParseSpec(spec []byte, filename string, schemas *terraform.Schemas) (*Spec, hcl.Diagnostics) {
-	type assert struct {
-		Type      string         `hcl:"type,label"`
-		Name      string         `hcl:"name,label"`
-		Config    hcl.Body       `hcl:",remain"`
-		DependsOn hcl.Expression `hcl:"depends_on,attr"`
+func ParseMockedMetadata(spec []byte, filename string) (*MockMetadata, hcl.Diagnostics) {
+	var r root
+	metadata := &MockMetadata{}
+	file, diags := hclparse.NewParser().ParseHCL(spec, filename)
+
+	if diags.HasErrors() {
+		return nil, diags
 	}
-	type mock struct {
-		Type   string   `hcl:"type,label"`
-		Name   string   `hcl:"name,label"`
-		Config hcl.Body `hcl:",remain"`
+	diags = gohcl.DecodeBody(file.Body, nil, &r)
+	if diags.HasErrors() {
+		return nil, diags
 	}
-	type root struct {
-		Asserts []*assert `hcl:"assert,block"`
-		Refutes []*assert `hcl:"refute,block"`
-		Mocks   []*mock   `hcl:"mock,block"`
-		// Modules   []*Module   `hcl:"module,block"`
+
+	metadata.Workspace = backend.DefaultStateName
+	for _, mockMetadata := range r.MockMetadata {
+		spec := hcldec.ObjectSpec{
+			"workspace": &hcldec.AttrSpec{
+				Name:     "workspace",
+				Type:     cty.String,
+				Required: false,
+			},
+		}
+		val, diags := hcldec.Decode(mockMetadata.Config, spec, nil)
+
+		if diags.HasErrors() {
+			return nil, diags
+		}
+
+		if !val.IsNull() {
+			workspace := val.GetAttr("workspace")
+			metadata.Workspace = workspace.AsString()
+		}
```

Comment:
You will always get the last element of the slice. If root.MockMetadata was not a slice, you could avoid this for loop

Reply:
Same question as above is there a way to make the mock_metadata optional via hcl if its not an array?